### PR TITLE
[Fixed] "Adding Flink on K8s Operator" 

### LINKED ISSUE #9134: Support Flink operator
**Description**

Airflow currently supports Spark operators for kicking off a spark-submit job.  In real-time computing or online machine learning scenarios, Flink operator could be designed to submit flink job in streaming workflow.

**Use case / motivation**

Airflow coud supports definition of `FlinkSubmitOperator` for DAG composed of multiple Flink jobs. `FlinkSubmitOperator` is designed to introduce the operator that kicks off a flink-submit task, and provides corresponding to JDBC and SQL operators.

**Related Issues**

N/A

### PR DESCRIPTION
**Pull request by xinbinhuang**
Port from: https://github.com/apache/airflow/pull/26664
closes: #9134 

Fixed provider metadata to make it pass CI

cc: @chethanuk-plutoflume, sorry for creating a new PR from your branch. I'm hoping to get this merged soon.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by xinbinhuang** [comment]
@potiuk I think the CI has some issue reporting back the error ([CI log](https://github.com/apache/airflow/actions/runs/3745083722/jobs/6359128624))

when I ran it locally, I got the traceback:

```Traceback (most recent call last):
  File "/Users/binh/.local/bin/breeze", line 8, in <module>
    sys.exit(main())
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/rich_click/rich_group.py", line 21, in main
    rv = super().main(*args, standalone_mode=False, **kwargs)
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/binh/.local/pipx/venvs/apache-airflow-breeze/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/commands/ci_commands.py", line 253, in selective_check
    print(str(sc), file=sys.stderr)
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/utils/selective_checks.py", line 275, in __str__
    output.append(get_ga_output(field_name, getattr(self, field_name)))
  File "/Users/binh/.pyenv/versions/3.9.11/lib/python3.9/functools.py", line 993, in __get__
    val = self.func(instance)
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/utils/selective_checks.py", line 558, in test_types
    current_test_types = set(self._get_test_types_to_run())
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/utils/selective_checks.py", line 539, in _get_test_types_to_run
    affected_providers = find_all_providers_affected(changed_files=self._files)
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/utils/selective_checks.py", line 237, in find_all_providers_affected
    add_dependent_providers(all_providers, provider, dependencies)
  File "/Users/binh/projects/airflow/dev/breeze/src/airflow_breeze/utils/selective_checks.py", line 223, in add_dependent_providers
    for dep_name in dependencies[provider_to_check]["cross-providers-deps"]:
KeyError: 'apache.flink'
```

Any clues if we can fix the CI to report back the traceback properly?

**Comment by potiuk** [comment]
Good one. Unforeseen interaction of stderr redirection to GITHUB_OUTPUT.

Fix in https://github.com/apache/airflow/pull/28514 (explanation in the commit descripton - and you can see the changes in the PR :)

File: airflow/providers/apache/flink/operators/flink_kubernetes.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        # default for in_cluster is True; patch if False and no KPO param.
+        conf_in_cluster = conf.getboolean("kubernetes", "in_cluster")
+        if self.in_cluster is None and conf_in_cluster is False:
+            hook._deprecated_core_in_cluster = conf_in_cluster  # noqa
+
+        # there's no default for cluster context; if we get something (and no KPO param) patch it.
+        conf_cluster_context = conf.get("kubernetes", "cluster_context", fallback=None)
+        if not self.cluster_context and conf_cluster_context:
+            hook._deprecated_core_cluster_context = conf_cluster_context  # noqa
+
```
Comment (xinbinhuang) [inline airflow/providers/apache/flink/operators/flink_kubernetes.py]: @chethanuk-plutoflume Are these actually needed ? Not see them being used/referenced anywhere.

**Comment by chethanuk-plutoflume** [comment]
Awesome thanks for fixing breeze cross-providers-deps @xinbinhuang

File: airflow/providers/apache/flink/operators/flink_kubernetes.py (hunk 1 above)
Comment (chethanuk) [reply airflow/providers/apache/flink/operators/flink_kubernetes.py]: https://github.com/apache/airflow/pull/28512/files#diff-5de054c1bea5c3b53b30769de1d806cd1d6075cc13732d984fbea70ab1501d6bR96
We are patching the hook

Comment (xinbinhuang) [reply airflow/providers/apache/flink/operators/flink_kubernetes.py]: Yeah I saw the function is being called, but I can't find the patched attributes being used anywhere (i.e. `hook._deprecated_core_cluster_context`)

Comment (chethanuk) [reply airflow/providers/apache/flink/operators/flink_kubernetes.py]: I think I was using _deprecated_core_cluster_context and old API in one of the internal tests, while I didn't open source. We can safely remove 119 and 120 in this case

Comment (xinbinhuang) [reply airflow/providers/apache/flink/operators/flink_kubernetes.py]: Got it. I think all these internal patching are not needed. I've removed them all

**Comment by xinbinhuang** [comment]
@potiuk @jedcunningham CI green. Wanna take a look?

**Review by kaxil — APPROVED** [review]

**Pull request merged**